src/grammar.pegjs

```diff
@@ -57,15 +57,24 @@ leaf_clause
     { return { pos: "leaf-clause", key: key, value: value }; }
 
 value
-  = operator_object
+  = operator_expression
   / JSON
 
 
 /* --- Operators --- */
 
-operator_object
+value_operator
+  = "$gte" / "$gt" / "$lte" / "$lt" / "$eq" / "$ne" / "$type" / "$size" / "$exists"
+
+list_operator
+  = "$in" / "$nin" / "$all"
+
+operator_expression_operator
+  = "$not" / "$elemMatch"
+
+operator_expression
```

Comment:
👍  Thanks for the [docs link](https://docs.mongodb.com/manual/reference/operator/query/not/index.html).